api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java

```diff
@@ -322,60 +314,84 @@ && hasPrivilege(userContext, candidate.getRequiredPrivilege())) {
 	}
 
 	@Override
-	public boolean checkRequireExpression(Requireable candidate, AppContextModel contextModel) {
-		try {
-			String requireExpression = candidate.getRequire();
-			if (StringUtils.isBlank(requireExpression)) {
-				return true;
-			} else {
-				// If any properties in contextModel are Maps, we want to allow scripts to access their subproperties
-				// with dot notation, but ScriptEngine and Bindings don't naturally handle this. Instead we will convert
-				// all Map-type properties to JSON and use this to define objects directly within the ScriptEngine.
-				// (Properties that are not Maps don't need this special treatment.)
-				ObjectMapper objectMapper = new ObjectMapper();
-				Map<String, Object> mapProperties = new HashMap<>();
-				for (Map.Entry<String, Object> e : contextModel.entrySet()) {
-					if (e.getValue() instanceof Map) {
-						mapProperties.put(e.getKey(), e.getValue());
-					}
-					else if(!"util".equals(e.getKey())) {
-						javascriptEngine.eval("var " + e.getKey() + " = " + objectMapper.writeValueAsString(e.getValue()) + ";");
-					}
-				}
+	public void addRequireExpressionScript(String key, String value) {
+		requireExpressionScripts.put(key, value);
+	}
```

Comment:
This method allows implementations to register new helper functions in their activators.